clients/ts/signalr/src/WebSocketTransport.ts

```diff
@@ -44,7 +47,22 @@ export class WebSocketTransport implements ITransport {
 
         return new Promise<void>((resolve, reject) => {
             url = url.replace(/^http/, "ws");
-            const webSocket = new this.webSocketConstructor(url);
+            let webSocket: WebSocket;
+            let cookies: string = "";
+            if ((this.httpClient as any).getCookies) {
+                cookies = (this.httpClient as any).getCookies(url);
+            }
+            if (cookies !== "") {
+                webSocket = new this.webSocketConstructor(url, undefined, {
```

Comment:
As with `EventSource`, can you only set the headers if we're not in the browser?